tests/test_relative_free_energy.py

```diff
@@ -169,6 +174,29 @@ def test_run_hif2a_test_system(estimate_relative_free_energy_fn):
             protein_path=str(protein_path),
             estimate_relative_free_energy_fn=estimate_relative_free_energy_fn,
         )
+
+
+@pytest.mark.nightly(reason="Slow!")
+@pytest.mark.parametrize(
+    "estimate_relative_free_energy_fn",
+    [
+        estimate_relative_free_energy,
+        estimate_relative_free_energy_bisection,
+        pytest.param(
+            estimate_relative_free_energy_bisection_hrex,
+            marks=pytest.mark.xfail(
+                reason="lambda window trajectories are not individually reproducible given InitialState due to mixing"
+            ),
+        ),
```

Comment:
The latter; this test simply samples from each window individually and compares the resulting frames against the results returned by `estimate_relative_free_energy_fn`. For HREX, the results are expected to be different because configurations are (deterministically, given a fixed seed) mixed between states during sampling.

The XFAIL here would fail if there is actually no mixing and the results end up being the same between HREX and individually sampling the states.

Reply:
👍 , thanks for clarifying!

>The XFAIL here would fail if there is actually no mixing and the results end up being the same between HREX and individually sampling the states.

Nice -- that's an added bonus.

Reply:
A test that verifies that calling `HREX` is deterministic/bitwise deterministic is important, but can't be replicated with the logic that is in this test.

`xfail` seems confusing here, as we know this shouldn't work and there are presumably better ways to assert that we get swapping, such as checking that the acceptance rate is at least > 0. I agree, the time to run this test that we know will fail seems bad.

Reply:
> A test that verifies that calling HREX is deterministic/bitwise deterministic is important

Yes, agreed we should have a test for bitwise determinism of HREX. Sorry, was stuck on thinking of something with the structure of this test, which verifies that the results of running individual simulations starting from each initial state match those returned by the `estimate_relative_free_energy` function. Something like this doesn't make sense for HREX, but we should still test end-to-end determinism of HREX, e.g. by running it twice in succession with the same seed.

Reply:
> nit: this is a strict xfail, so the test itself will fail if the assertions pass. So I would argue it does provide some signal, but maybe not worth the time required to run it.

Thinking about this some more, this will always fail. Even if there are no swaps. Because of the way you run the equilibration in the hrex it is `eq_steps + 1` rather than `eq_steps` so even calling `sample()` with the original MDParams will produce a difference. I get your point, this simply seems to be too abstract of a test that requires thinking about running HREX in an atypical way.

Reply:
> Because of the way you run the equilibration in the hrex it is eq_steps + 1 rather than eq_steps so even calling sample() with the original MDParams will produce a difference.

Yes, good point. Will just remove this test.

Reply:
Changed from `xfail` to `skip` in https://github.com/proteneer/timemachine/pull/1128/commits/ccbcb619501c32e7eca6adc39e612c6335dfbf88.

(Skipped instead of removed for documentation purposes)

Reply:
Added a test for reproducibility in https://github.com/proteneer/timemachine/pull/1128/commits/d0f5a030e49958ef0af726341835278cfbefcfe4